components/runes/src/db/cache/rune_validation.rs

```diff
@@ -0,0 +1,686 @@
+async fn rune_etching_has_valid_commit(
+    bitcoin_client: &BitcoinRPCClient,
+    ctx: &Context,
+    tx: &Transaction,
+    rune: &Rune,
+    reveal_block_height: u32,
+    inputs_counter: &mut u64,
+) -> Result<bool, Box<dyn std::error::Error + Send + Sync>> {
+    let commitment = rune.commitment();
+
+    // Check each input for valid commitment (the commitment could be in any input)
+    for input in &tx.input {
+        *inputs_counter += 1;
+        // Extract tapscript from witness data (note: this doesn't guarantee the spent output is taproot)
+        let Some(tapscript) = unversioned_leaf_script_from_witness(&input.witness) else {
+            continue;
+        };
+
+        for instruction in tapscript.instructions() {
+            // Skip invalid script instructions
+            let Ok(instruction) = instruction else {
+                break;
+            };
+
+            let Some(pushbytes) = instruction.push_bytes() else {
+                continue;
+            };
+
+            // Check if this instruction pushes the expected commitment bytes
+            if pushbytes.as_bytes() != commitment {
+                continue;
+            }
+
+            let txid =
+                bitcoincore_rpc::bitcoin::Txid::from_str(&input.previous_output.txid.to_string())
+                    .unwrap();
+
+            // Fetch the commit transaction to validate taproot output and get block height
+            let Some(commit_tx_info) = bitcoin_get_raw_transaction(bitcoin_client, ctx, &txid)
+            else {
+                panic!(
+                    "can't get input transaction: {}",
+                    input.previous_output.txid
+                );
+            };
+
+            // Verify the spent output is a taproot (P2TR) output
+            let taproot = commit_tx_info.vout[input.previous_output.vout as usize]
+                .script_pub_key
+                .script()?
+                .is_p2tr();
+
+            if !taproot {
+                continue;
+            }
+
+            // Get commit transaction's block height to check confirmation count
+            let commit_tx_height =
+                bitcoind_get_block_height(bitcoin_client, ctx, &commit_tx_info.blockhash.unwrap());
+
+            // Calculate confirmations and check minimum requirement (6 blocks)
+            let confirmations = reveal_block_height.checked_sub(commit_tx_height).unwrap() + 1;
+            if confirmations >= u32::from(Runestone::COMMIT_CONFIRMATIONS) {
+                return Ok(true);
+            }
+        }
+    }
+
+    Ok(false)
+}
+
+/// Validates that a rune etching transaction has a proper commitment transaction with automatic reconnection.
+///
+/// This function wraps the core validation logic with connection error handling. If a connection
+/// error is detected, it will attempt to reconnect the Bitcoin RPC client and retry the validation.
+pub async fn rune_etching_has_valid_commit_with_reconnect(
```

Comment:
Same here, I don't think we need reconnect logic at this level. Is there a special reason why you're doing it here? Maybe I'm missing something